File: common/src/main/java/com/demo/common/dto/TaskPhaseName.java
package com.demo.common.dto;

/**
 * Fixed set of phase names available in a project's workflow.
 * Using an enum prevents arbitrary strings and keeps phase names consistent across services.
 */
public enum TaskPhaseName {
    PLANNING,
    BACKLOG,
    TODO,
    IN_PROGRESS,
    IN_REVIEW,
    TESTING,
    DONE,
    RELEASED,
    REJECTED
}

File: android/data/src/main/kotlin/com/demo/taskmanager/data/dto/enums/TaskPhaseName.kt
package com.demo.taskmanager.data.dto.enums

import kotlinx.serialization.Serializable

@Serializable
enum class TaskPhaseName {
    PLANNING,
    BACKLOG,
    TODO,
    IN_PROGRESS,
    IN_REVIEW,
    TESTING,
    DONE,
    RELEASED,
    REJECTED,
}
